driver/driver_test.go

```diff
@@ -861,3 +755,30 @@ func createContrailNetwork(c *controller.Controller) *types.VirtualNetwork {
 	return controller.CreateMockedNetworkWithSubnet(
 		c.ApiClient, networkName, subnetCIDR, project)
 }
+
+func deleteTheOnlyHNSEndpoint(d *ContrailDriver) {
+	_, hnsEndpointID := getTheOnlyHNSEndpoint(d)
+	err := hns.DeleteHNSEndpoint(hnsEndpointID)
+	Expect(err).ToNot(HaveOccurred())
+}
+
+func getTheOnlyHNSEndpoint(d *ContrailDriver) (*hcsshim.HNSEndpoint, string) {
+	hnsNets, err := contrailDriver.hnsMgr.ListNetworks()
+	Expect(err).ToNot(HaveOccurred())
+	eps, err := hns.ListHNSEndpointsOfNetwork(hnsNets[0].Id)
+	Expect(err).ToNot(HaveOccurred())
+	hnsEndpointID := eps[0].Id
+	hnsEndpoint, err := hns.GetHNSEndpoint(hnsEndpointID)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(hnsEndpoint).ToNot(BeNil())
+	return hnsEndpoint, hnsEndpointID
+}
+
+func setupNetworksAndEndpoints(c *controller.Controller, docker *dockerClient.Client) (
+	*types.VirtualNetwork, string, string) {
+	contrailNet := createContrailNetwork(c)
+	dockerNetID := createValidDockerNetwork(docker)
+	containerID, err := runDockerContainer(docker)
```

Comment:
The only reason I'm creating containers is so that docker daemon sends a request to the driver. We don't care if it's nanoserver, servercore or anything else. I don't want to test docker itself.